packages/kiota_abstractions/lib/src/case_insensitive_map.dart

```diff
@@ -109,6 +109,7 @@ class CaseInsensitiveMap<K extends String, V> implements Map<K, V> {
 
   @override
   V putIfAbsent(K key, V Function() ifAbsent) {
+    _originalKeys[normalizeKey(key)] = key;
     return _contents.putIfAbsent(normalizeKey(key), ifAbsent);
```

Comment:
nit: only call `normalizeKey(key)` once and only modify the list if actually absent:

```suggestion
    final normalizedKey = normalizeKey(key);

    return _contents.putIfAbsent(normalizedKey, () {
        _originalKeys[normalizedKey] = key;

        return ifAbsent()
    });
```